### PR TITLE
refactor(api): remove heater shaker feature flag

### LINKED ISSUE #10775: api: remove heater-shaker feature flag
## Overview

As a user, I'd like to use a Heater-Shaker module in my Python protocols. In order to do that, we need to promote this feature to production by removing the robot-side feature-flag that currently guards heater-shaker functionality.

## Acceptance criteria

- [ ] Loading and using a Heater-Shaker in a Python protocol works
- [ ] `GET /settings` does not report any heater shaker feature flag

## Implementation notes

- The `enableHeaterShakerPAPI` feature flag should be removed from `opentrons.config` module
    - A new config file migration must be created
    - Implement by following the conventions of existing migrations
- In the codebase, check for calls to `enable_heater_shaker_python_api()`
    - Promote the body of any `if` statement guarded by the boolean to the default behavior

### PR DESCRIPTION
**Pull request by TamarZanzouri**
# Overview

closes #10775.
remove heater shaker feature flag.

# Changelog

- Removed heater shaker advance setting.
- Added a migration function
- Remove\updated related tests 

# Review requests

are all feature flag changes removed? 

# Risk assessment

small. heater shaker should now be available without setting the feature flag.

### CODE REVIEW

File: api/tests/opentrons/protocol_runner/smoke_tests/test_legacy_module_commands.py [hunk 1]
```diff
@@ -52,7 +52,6 @@ def run(ctx):
 
 async def test_runner_with_modules_in_legacy_python(
     modules_legacy_python_protocol_file: Path,
-    enable_heater_shaker_python_api: None,
```
Comment (mcous) [inline api/tests/opentrons/protocol_runner/smoke_tests/test_legacy_module_commands.py:55]: Can we make sure we remove this fixture from `conftest.py`, too?

File: api/src/opentrons/config/advanced_settings.py [hunk 2]
```diff
@@ -460,6 +452,16 @@ def _migrate14to15(previous: SettingsMap) -> SettingsMap:
     return newmap
 
 
+def _migrate15to16(previous: SettingsMap) -> SettingsMap:
+    """Migrate to version 16 of the feature flags file.
+
+    - Removes deprecated enableProtocolEngine option
```
Comment (SyntaxColoring) [inline api/src/opentrons/config/advanced_settings.py]: Looks like a copy-paste mistake?

```suggestion
    - Removes deprecated enableHeaterShakerPAPI option
```

**Review by sanni-t — APPROVED** [review]
🏁

File: api/src/opentrons/protocol_api/protocol_context.py [hunk 3]
```diff
@@ -487,13 +485,6 @@ def load_module(
         if not load_result:
             raise RuntimeError(f"Could not find specified module: {module_name}")
 
-        # TODO(mc, 2022-06-14): remove guard for heater-shaker production release
-        if (
-            load_result.type == ModuleType.HEATER_SHAKER
-            and not ff.enable_heater_shaker_python_api()
-        ):
-            raise UnsupportedAPIError("Heater-Shaker module is not yet supported")
-
```
Comment (SyntaxColoring) [inline api/src/opentrons/protocol_api/protocol_context.py:496]: 1. Does this PR also need to resolve the docstring `# TODO` above?

   ```python
    # TODO: add heater-shaker to the returns values in above docstring
    ```
2. We need to reject Heater-Shaker loads in protocols whose `apiLevel` is too old, similarly to how we handle the Thermocycler, above on line 476. I don't think we're currently doing that anywhere. Should this PR do that? If so, it probably also warrants a test.

**Review by SyntaxColoring — CHANGES_REQUESTED** [review]

File: api/src/opentrons/protocol_api/protocol_context.py (hunk 3 above)
Comment (ecormany) [reply api/src/opentrons/protocol_api/protocol_context.py:496]: Good catch, Max. All of the other changes that will show up on the API Version 2 Reference page in docs are in #10686, so maybe it's best to make this change there.

Comment (SyntaxColoring) [reply api/src/opentrons/protocol_api/protocol_context.py:496]: Discussed on a call with @sanni-t and @TamarZanzouri:

1. Like @ecormany suggested above, we'll update the docstring in #10686.
2. It turns out we don't have that check for any other module. A user can technically load a Thermocycler in an `"apiLevel": "2.0"` protocol as long as they don't provide the `configuration` parameter; they just can't do anything with the returned `ThermocyclerContext` object. I think we should probably add this check for all modules, but that's not a problem for this PR.

**Review by SyntaxColoring — APPROVED** [review]

**Pull request merged**